### ISSUE TITLE
Takahē posts viewed in clients open hashtag as URL

### ISSUE THREAD

**Issue by chdorner** [jointakahe/takahe#541]
This is related to #448 but not exactly the same.

When directly logged in to Takahē with certain clients and I click on a hashtag of one of my own posts, those clients open the browser with the hashtag link.

So far I can see this happening with:
- Ivory
- Mona
- Semaphore / Pinafore
- Elk

Additional clients may be affected too.

IceCubes is the only client I saw it working correctly

**Comment by andrewgodwin** [comment]
That was in fact what happened with #448 - the hashtags were parsed as links by clients.

**Comment by chdorner** [comment]
Right, I understood #448 to be specifically for clients talking to a proper Mastodon API retrieving a Takahē status.

I'm seeing the same bug happening with these clients but when they talk directly to the Takahē API

**Comment by andrewgodwin** [comment]
It's all about how the links are formatted - the previous fix was to have `rel="tag"` in there. If that's gone it's definitely a regression.

**Comment by chdorner** [comment]
It does have `rel="tag"`

this is the JSON according to `/debug/json`:

```js
{
    // ...
    "content": "<p><a href=\"https://social.chdorner.com/tags/lightroom/\" rel=\"tag\">#Lightroom</a>, go home you&#x27;re drunk.</p>",
    "tag": [
        {
            "href": "https://social.chdorner.com/tags/lightroom/",
            "name": "#lightroom",
            "type": "Hashtag"
        }
    ],
    // ...
}
```

the only difference I can see from an example originating from Mastodon is the `class="mention hashtag"`, in i.e.: `<p><a href="https://vmst.io/tags/mastoadmin" class="mention hashtag" rel="tag">#<span>mastoadmin</span></a></p>`
from [this](https://vmst.io/@vmstan/110027699424938735) status.

**Comment by chdorner** [comment]
Found the Elk parser: https://github.com/elk-zone/elk/blob/dc397e133c9009ea8f6a014cf453cc602d3c96eb/composables/content-render.ts#L83-L103

Almost seems like it's ignoring the `tags` field returned by the Mastodon API entirely.
Seems that it goes through the content field and checks all anchor tags with a class `"mention"`, then match against `/^https?:\/\/([^/]+)\/tags\/([^/]+)$/` and use the latest match group as the tag name.

Unclear how other clients implement this.

**Comment by chdorner** [comment]
Semaphore / Pinafore has a similar logic, starting with all anchors having the `hashtag` class, and then matching them up against what's in the `tags` field coming from the Mastodon API.

https://github.com/NickColley/semaphore/blob/a7eaec339186330d402c67cebdd1fce110dc0155/src/routes/_components/status/StatusContent.html#L120-L130

**Comment by chdorner** [comment]
Fix for Ivory and Mona is in #542 

Elk and Semaphore are still failing because our hashtag URLs end with a `/`

**Comment by chdorner** [comment]
Closing this as there's not much we can do for Semaphore and Elk given that they parse the URL and are requiring them to not have trailing slashes
